### ISSUE TITLE
Bug - is not validating correctly when destroying/recreating input

### ISSUE THREAD

**Issue by ileonardo-c** [pablo-abc/felte#23]
Editing for the use of email to better understand, but basically the validation is lost when it destroys / recreates the `input`

- I have two inputs. Each one shows according to a previous selection.
- If user selects `Paypal`, I need `email-paypal`
- If user selects `Payoneer, I need `email-payoneer`
- Each input is only shown if needed. We don't show `email-paypal` if user selected `Payoneer`
- Both inputs are linked to `email` variable – so, if user types in one input, it will be applied to the another one

The problem is:

1. Select `Paypal`
2. Enter an *invalid* value in `email-paypal`
3. See the error message
4. Select `Payoneer`
5. Type a *valid* value in `email-payoneer`
6. Select `Paypal` again
7. Note that now the input value is VALID, but the error message is still showing up
8. Click on `Submit` – it won't submit, saying that the `email-paypal` is invalid, but it's not
9. Type a new character in `email-paypal`
10. The error message will disappear and you can submit

Basically, when an input is destroyed, it's reference continues in `$error` store and when it is created again it's already showing as "invalid".

Example: https://codesandbox.io/s/bug-validate-if-felte-forked-b0kgj?file=/App.svelte:192-197
Video: https://user-images.githubusercontent.com/15247903/115600317-f7311700-a2b2-11eb-88f5-d4c31479afd7.mp4

**Comment by pablo-abc** [comment]
The reproduction was really useful! Thanks a lot.

I'm not sure if this can be considered a bug after testing it. This is happening because Felte is handling the following data:

```javascript
{
  "wallet": "payoneer",
  "email-paypal": "",
  "email-payoneer": ""
}
```

While you're binding your "real" value to an external variable `email` that Felte is not aware of. There's nothing linking the `email` var to `email-paypal` or `email-payoneer`.

If you make a relation explicit is does work, e.g. this fixes it for me:

```javascript
$: {
  $data["email-paypal"] = email;
  $data["email-payoneer"] = email;
}
```

Which makes it so every time `email` changes, you also update those fields.

But if we're going this route, and the distinction for the email is already contained within the `wallet` prop of `data`, why not use a form like this?

```html
<div style="margin-bottom: 20px">
  <div>Wallet:</div>
  <input
    type="radio"
    id="wallet-paypal"
    name="wallet"
    value="paypal"
    bind:group={wallet}
   >
  <label for="wallet-paypal">Paypal</label><br>
  <input
    type="radio"
    id="wallet-payoneer"
    name="wallet"
    value="payoneer"
    bind:group={wallet}
  >
  <label for="wallet-payoneer">Payoneer</label><br>
</div>
<div>Your <strong>{wallet}</strong> email:</div>
<input
  type="text"
  id="email"
  name="email"
  aria-describedby="email-validation"
 >
<div id="email-validation" data-felte-reporter-dom-for="email" />
```

Unless there's a use-case I'm not seeing, I can't seem to see why changing `inputs` is necessary for this. 😅

**Comment by pablo-abc** [comment]
I'm closing this since there seems to be no work to do here. If you have an objection to this feel free to reopen this!
